streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -1198,6 +1198,48 @@ public void shouldCloseActiveTasksAndPropagateThreadProducerExceptionsOnCleanShu
         verify(activeTaskCreator, changeLogReader);
     }
 
+    @Test
+    public void shouldCloseActiveTasksDirtyAndPropagateCommitException() {
+        final StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(
+            new Metrics(), "clientId", StreamsConfig.METRICS_LATEST);
+        taskManager = new TaskManager(
+            changeLogReader,
+            UUID.randomUUID(),
+            "taskManagerTest",
+            streamsMetrics,
+            activeTaskCreator,
+            standbyTaskCreator,
+            topologyBuilder,
+            adminClient,
+            stateDirectory,
+            StreamThread.ProcessingMode.EXACTLY_ONCE_ALPHA
+        );
+        taskManager.setMainConsumer(consumer);
+
+        final Task task00 = new StateMachineTask(taskId00, taskId00Partitions, true);
+        final StateMachineTask task01 = new StateMachineTask(taskId01, taskId01Partitions, true);
```

Comment:
nit: why use `Task` for `task00` but `StateMachineTask` for `task01` ? Can you unify this (it a little confusing)?

Reply:
Because we don't have `setCommitNeeded` for `Task` I guess?